wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -325,53 +321,78 @@ public static void main(String[] args) {
 
     @Override
     public Integer call() throws IOException, BlockStoreException {
-        ActionEnum action;
-        try {
-            action = ActionEnum.valueOf(actionStr.toUpperCase().replace("-", "_"));
-        } catch (IllegalArgumentException e) {
-            System.err.println("Could not understand action name " + actionStr);
-            return 1;
-        }
+        ActionEnum action = validateAction(actionStr);
+        if (action == null) return 1;
 
-        if (debugLog) {
-            BriefLogFormatter.init();
-            log.info("Starting up ...");
-        } else {
-            // Disable logspam unless there is a flag.
-            java.util.logging.Logger logger = LogManager.getLogManager().getLogger("");
-            logger.setLevel(Level.SEVERE);
-        }
+        setupLogging();
         params = NetworkParameters.of(net);
-        String fileName = String.format("%s.chain", net);
-        if (chainFile == null) {
-            chainFile = new File(fileName);
-        }
+        setDefaultChainFile();
         Context.propagate(new Context());
 
-        if (conditionStr != null) {
-            condition = new Condition(conditionStr);
-        }
+        initializeCondition(conditionStr);
 
         if (action == ActionEnum.CREATE) {
             createWallet(net, walletFile);
             return 0;  // We're done.
         }
-        if (!walletFile.exists()) {
-            System.err.println("Specified wallet file " + walletFile + " does not exist. Try wallet-tool --wallet=" + walletFile + " create");
-            return 1;
+        if (checkWalletExists() == 1) return 1;
+
+        Integer walletResult = initializeWallet(action);
+        if (walletResult != 0) return walletResult; //return on failure
+
+        Integer networkCheckResult = checkWalletNetwork();
+        if (networkCheckResult != 0) return networkCheckResult;//return on failure
+
+        Integer actionResult = processAction(action);
+        if (actionResult != 0) return actionResult; //return on failure
+
+        Integer walletConsistencyResult =  checkWalletConsistency();
+        if(walletConsistencyResult != 0) return walletConsistencyResult;
+        saveWallet(walletFile);
+
+        if (waitFor != null) {
+            Integer waitForResult = handleWaitForCondition();
+            if (waitForResult != 0) return waitForResult;
         }
+        shutdown();
 
-        if (action == ActionEnum.RAW_DUMP) {
-            // Just parse the protobuf and print, then bail out. Don't try and do a real deserialization. This is
-            // useful mostly for investigating corrupted wallets.
-            try (FileInputStream stream = new FileInputStream(walletFile)) {
-                Protos.Wallet proto = WalletProtobufSerializer.parseToProto(stream);
-                proto = attemptHexConversion(proto);
-                System.out.println(proto.toString());
-                return 0;
-            }
```

Comment:
I think the RAW_DUMP is handled early so that it can be printed even if some of the consistency checks fail.  Moving RAW_DUMP to a method is a good idea, but moving it into `processAction` will prevent people from using this RAW_DUMP to dump/debug a corrupted wallet. If you could make just the move to a function a separate commit that would be great.